multihash_iterator.go

```diff
@@ -5,10 +5,15 @@ import (
 	"io"
 	"sort"
 
+	"github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
 	carindex "github.com/ipld/go-car/v2/index"
+	hamt "github.com/ipld/go-ipld-adl-hamt"
+	"github.com/ipld/go-ipld-prime"
 	"github.com/multiformats/go-multihash"
 )
 
+var _ MultihashIterator = (*sliceMhIterator)(nil)
```

Comment:
nit: This is not necessary since `SliceMultihashIterator` returns `&sliceMhIterator` as a `MultihashIterator` interface, thereby guaranteeing that `sliceMhIterator` implements `MultihashIterator`.

This check is only needed when the constructor returns the concrete type or a different interface type from the target interface.

Maybe better to return the concrete type, and let the calling code pass it around as a `MultihashIterator`?